.github/workflows/wpiformat.yml

```diff
@@ -0,0 +1,34 @@
+name: Lint and Format
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  wpiformat:
+    name: "wpiformat"
```

Comment:
This all seems rather verbose and even a bit excessive for just linting, although I acknowledge it [came from FIRST](https://docs.wpilib.org/en/stable/docs/software/advanced-gradlerio/code-formatting.html). I recommend using the `cpp-linter` workflow [found here instead](https://github.com/cpp-linter/cpp-linter-action) for most of the action. This way, we can have people [install a formatting extension](https://code.visualstudio.com/docs/cpp/cpp-ide#_code-formatting) if they want which can still read from the `.clang-format` file while integrating nicely with GitHub. Feel free to continue the discussion with your ideas, however.